spring-transaction/src/test/kotlin/org/jetbrains/exposed/spring/ExposedTransactionManagerTest.kt

```diff
@@ -87,45 +96,162 @@ open class ExposedTransactionManagerTest : SpringTransactionTestBase() {
+            T1.insertRandom()
             Assert.assertEquals(2, T1.selectAll().count())
         }
-        SchemaUtils.drop(T1)
+        Assert.assertEquals(2, T1.selectAll().count())
     }
 
     @Test
     @Repeat(5)
-    fun testConnectionWithoutAnnotation() {
-        transaction {
-            SchemaUtils.create(T1)
-            val rnd = Random().nextInt().toString()
-            T1.insert {
-                it[c1] = rnd
+    @Transactional
+    open fun testConnectionWithNestedTransactionInnerRollback() {
+        T1.insertRandom()
+        Assert.assertEquals(1, T1.selectAll().count())
+        transactionManager.execute(TransactionDefinition.PROPAGATION_NESTED) { status ->
+            T1.insertRandom()
+            Assert.assertEquals(2, T1.selectAll().count())
+            status.setRollbackOnly()
+        }
+        Assert.assertEquals(1, T1.selectAll().count())
+    }
+
+    @Test
+    @Repeat(5)
+    fun testConnectionWithNestedTransactionOuterRollback() {
+        transactionManager.execute {
+            T1.insertRandom()
+            Assert.assertEquals(T1.selectAll().count(), 1)
+            it.setRollbackOnly()
+
+            transactionManager.execute(TransactionDefinition.PROPAGATION_NESTED) {
+                T1.insertRandom()
+                Assert.assertEquals(T1.selectAll().count(), 2)
             }
+            Assert.assertEquals(T1.selectAll().count(), 2)
+        }
 
-            Assert.assertEquals(T1.selectAll().single()[T1.c1], rnd)
+        transactionManager.execute {
+            Assert.assertEquals(T1.selectAll().count(), 0)
         }
-        assertNull(TransactionManager.currentOrNull())
-        transaction {
-            val rnd = Random().nextInt().toString()
-            T1.insert {
-                it[c1] = rnd
+    }
+
+    @Test
+    @Repeat(5)
+    @Transactional
+    open fun testConnectionWithRequiresNew() {
+        T1.insertRandom()
+        Assert.assertEquals(T1.selectAll().count(), 1)
+        transactionManager.execute(TransactionDefinition.PROPAGATION_REQUIRES_NEW) {
+            Assert.assertEquals(T1.selectAll().count(), 0)
+            T1.insertRandom()
+            Assert.assertEquals(T1.selectAll().count(), 1)
+        }
+        Assert.assertEquals(T1.selectAll().count(), 2)
+    }
+
+    @Test
+    @Repeat(5)
+    fun testConnectionWithRequiresNewWithInnerTransactionRollback() {
+        transactionManager.execute {
+            T1.insertRandom()
+            Assert.assertEquals(T1.selectAll().count(), 1)
+            transactionManager.execute(TransactionDefinition.PROPAGATION_REQUIRES_NEW) {
+                T1.insertRandom()
+                Assert.assertEquals(T1.selectAll().count(), 1)
+                it.setRollbackOnly()
             }
+            Assert.assertEquals(T1.selectAll().count(), 1)
+        }
 
-            Assert.assertEquals(2, T1.selectAll().count())
+        transactionManager.execute {
+            Assert.assertEquals(T1.selectAll().count(), 1)
```

Comment:
Just a minor point about the 3 tests above, but it's good for consistency throughout the codebase:

Please swap the arguments to always follow the convention `Assert.assertEquals(expected, actual)`, so:

`Assert.assertEquals(1, T1.selectAll().count())`

Reply:
I made this mistake because I had never used junit before. Thanks for pointing it out :) 

Would it be correct to change the existing rnd value test to `Assert.assertEquals(rnd, T1.selectAll().single()[T1.c1])` as well?

Reply:
@FullOfOrange Yes please, the old tests that don't follow the convention `(expected, actual)` can be changed too.